File: ClickServer/generatedJava/com/pku/ss/egl/libs/HotKeysControllerLib.java
package com.pku.ss.egl.libs;
import org.eclipse.edt.javart.resources.*;
import org.eclipse.edt.javart.*;
import eglx.lang.SysLib;
import org.eclipse.edt.runtime.java.eglx.lang.EAny;
import java.lang.Object;
import eglx.lang.AnyException;
import com.pku.ss.egl.et.HotKeys;
@SuppressWarnings("unused")
@javax.xml.bind.annotation.XmlRootElement(name="HotKeysControllerLib")
public class HotKeysControllerLib extends ExecutableBase {
	private static final long serialVersionUID = 10L;
	private transient HotKeys hk;
	public HotKeysControllerLib() {
		super();
	}
	{
		ezeInitialize();
	}
	public void ezeInitialize() {
		hk = new HotKeys();
	}
	public HotKeys getHk() {
		return hk;
	}
	public void setHk(HotKeys ezeValue) {
		hk = ezeValue;
	}
	public void left() {
		try {
			HotKeys.left();
		}
		catch ( java.lang.Exception eze$Temp1 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp1 );
			AnyException e;
			if ( eze$Temp1 instanceof AnyException ) {
				e = (AnyException)eze$Temp1;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp1);
			}
			{
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
	public void right() {
		try {
			HotKeys.right();
		}
		catch ( java.lang.Exception eze$Temp2 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp2 );
			AnyException e;
			if ( eze$Temp2 instanceof AnyException ) {
				e = (AnyException)eze$Temp2;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp2);
			}
			{
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
	public void backwardCmd() {
		try {
			HotKeys.backward();
		}
		catch ( java.lang.Exception eze$Temp3 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp3 );
			AnyException e;
			if ( eze$Temp3 instanceof AnyException ) {
				e = (AnyException)eze$Temp3;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp3);
			}
			{
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
	public void forwardCmd() {
		try {
			HotKeys.forward();
		}
		catch ( java.lang.Exception eze$Temp4 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp4 );
			AnyException e;
			if ( eze$Temp4 instanceof AnyException ) {
				e = (AnyException)eze$Temp4;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp4);
			}
			{
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
}

File: ClickServerWeb/WebContent/com/pku/ss/egl/libs/HotKeysControllerLib.js
define([],function(){
	egl.defineRUILibrary('com.pku.ss.egl.libs', 'HotKeysControllerLib',
	{
		'eze$$fileName': 'com/pku/ss/egl/libs/HotKeysControllerLib.egl',
		'eze$$runtimePropertiesFile': 'com.pku.ss.egl.libs.HotKeysControllerLib',
			"constructor": function() {
				if(egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst']) return egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst']=this;
			}
			,
			"eze$$setEmpty": function() {
				this.hk = new ();
			}
			,
			"eze$$setInitial": function() {
				if(egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst']) return egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.HotKeysControllerLib['$inst']=this;
				this.eze$$setEmpty();
			}
			,
			"eze$$getAnnotations": function() {
				if(this.annotations === undefined){
					this.annotations = {};
					this.annotations["XMLRootElement"] = new egl.eglx.xml.binding.annotation.XMLRootElement("HotKeysControllerLib", null, false);
				}
				return this.annotations;
			}
			,
			"eze$$getFieldInfos": function() {
				if(this.fieldInfos === undefined){
					var annotations;
					this.fieldInfos = new Array();
				}
				return this.fieldInfos;
			}
			,
			"left": function() {
				try {
					egl.com.pku.ss.egl.et.HotKeys.left();
				}
				catch ( eze$Temp1 )
				{
					{
						if (!(eze$Temp1 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp1 = egl.makeExceptionFromCaughtObject(eze$Temp1);
						}
						var e = eze$Temp1;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"right": function() {
				try {
					egl.com.pku.ss.egl.et.HotKeys.right();
				}
				catch ( eze$Temp2 )
				{
					{
						if (!(eze$Temp2 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp2 = egl.makeExceptionFromCaughtObject(eze$Temp2);
						}
						var e = eze$Temp2;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"backwardCmd": function() {
				try {
					egl.com.pku.ss.egl.et.HotKeys.backward();
				}
				catch ( eze$Temp3 )
				{
					{
						if (!(eze$Temp3 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp3 = egl.makeExceptionFromCaughtObject(eze$Temp3);
						}
						var e = eze$Temp3;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"forwardCmd": function() {
				try {
					egl.com.pku.ss.egl.et.HotKeys._forward();
				}
				catch ( eze$Temp4 )
				{
					{
						if (!(eze$Temp4 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp4 = egl.makeExceptionFromCaughtObject(eze$Temp4);
						}
						var e = eze$Temp4;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"getHk": function() {
				return hk;
			}
			,
			"setHk": function(ezeValue) {
				this.hk = ezeValue;
			}
			,
			"toString": function() {
				return "[HotKeysControllerLib]";
			}
		}
	);
});
